src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -51,6 +51,149 @@ public class C
             Assert.True(publicProperty.SetMethod.IsInitOnly);
         }
 
+        [Theory, CombinatorialData, WorkItem(50245, "https://github.com/dotnet/roslyn/issues/50245")]
+        public void TestCSharp8_ConsumptionInObjectInitializer(bool useMetadataImage)
```

Comment:
>TestCSharp8_ConsumptionInObjectInitializer [](start = 20, length = 42)

Consider testing a scenario when the same C# 8 compilation defines init-only property and attempts to use it. In this case, I think, we should suppress the error at the use-site.

Reply:
I wasn't sure the right way to suppress the error in source-only scenario without suppressing in compilation-reference scenario. I ended up comparing compilations. Is there a better way?

---
In reply to: [552033954](https://github.com/dotnet/roslyn/pull/50251#discussion_r552033954) [](ancestors = 552033954)

Reply:
>  I ended up comparing compilations. Is there a better way?

This is exactly the kind of check I had in mind.

---
In reply to: [552272345](https://github.com/dotnet/roslyn/pull/50251#discussion_r552272345) [](ancestors = 552272345,552033954)